src/api/ci/jenkins/services/jenkins-build.service.ts

```
@@ -0,0 +1,377 @@
+
+      console.log(`Found ${jobInfo.builds.length} builds, checking up to ${maxBuildsToCheck}`);
+
+      // Limit the number of builds to check
+      const buildsToCheck = jobInfo.builds.slice(0, maxBuildsToCheck);
+      const matchingBuilds: JenkinsBuild[] = [];
+
+      // Check each build for the commit SHA
+      for (const buildRef of buildsToCheck) {
+        console.log(`Checking build #${buildRef.number}`);
+        const buildInfo = await this.getBuild(jobName, buildRef.number, folderName, false);
+        
```

ESLint

[error] Delete `········`

(prettier/prettier)